### PR TITLE
Add rpc to set log level

### LINKED ISSUE #5: Add service to change the log level of a dogu

### PR DESCRIPTION
**Pull request by nfranzeck**
Resolve #5

### CODE REVIEW

File: grpc-protobuf/logging.proto [hunk 1]
```diff
@@ -11,6 +12,9 @@ service DoguLogMessages {
   // QueryForDogu queries logs for a dogu. This api-call is only available in a MultiNode CES with k8s-ces-control.
   // The logs are queried from k8s-loki and therefore it is only possible to query logs from period of max 30 days.
   rpc QueryForDogu(DoguLogMessageQueryRequest) returns (stream DoguLogMessage);
+
+  // SetLogLevel sets the log level for a specific dogu.
+  rpc SetLogLevel(LogLevelRequest) returns (google.protobuf.Empty);
```
Comment (ppxl) [inline grpc-protobuf/logging.proto:17]: I was just informed that `SetLogLevel` not only sets a value but also restarts a dogu. This breaks the principle of least surprise given the doc comment and the method name.

I highly suggest to change both to reflect the possible downtime associated with a call. 

Maybe there would be another change be necessary, namely: If the default log level is in place (that is: no value was set) and the administrator sets the loglevel to a value equal to the loglevel. Under this circumstance the dogu's downtime could be avoided.

A rough idea of how the nature of dogu restart after a loglevel change could be reflected in the protobuf:

```suggestion
  // ApplyLogLevelWithRestart sets the log level for a specific dogu and restarts the dogu if the log level was changed.
  rpc ApplyLogLevelWithRestart(LogLevelRequest) returns (google.protobuf.Empty);
```

File: grpc-protobuf/logging.proto [hunk 2]
```diff
@@ -40,4 +44,18 @@ message DoguLogMessage {
   google.protobuf.Timestamp timestamp = 1;
   // message contains the content of logged message
   string message = 2;
+}
+
+enum LogLevel {
```
Comment (ppxl) [inline grpc-protobuf/logging.proto:49]: Basically, a protobuf describes an API. All things exported should contain a doc comment. Given a sufficient description some exported symbols can be left out, f. i. if `LogLevel` is written like so, I would not add comments on each log level value:

```suggestion
// LogLevel defines values which control the logging in decreasing verbosity.
enum LogLevel {
```

File: grpc-protobuf/logging.proto [hunk 3]
```diff
@@ -11,6 +12,9 @@ service DoguLogMessages {
   // QueryForDogu queries logs for a dogu. This api-call is only available in a MultiNode CES with k8s-ces-control.
   // The logs are queried from k8s-loki and therefore it is only possible to query logs from period of max 30 days.
   rpc QueryForDogu(DoguLogMessageQueryRequest) returns (stream DoguLogMessage);
+
+  // SetLogLevel sets the log level for a specific dogu.
+  rpc SetLogLevel(LogLevelRequest) returns (google.protobuf.Empty);
 }
 
 message DoguLogMessageRequest {
```
Comment (ppxl) [inline grpc-protobuf/logging.proto:20]: Like above, I suggest to add doc comments also on top level symbols.

**Review by ppxl — COMMENTED** [review]
Thanks for the work :muscle: Looking good but I found some medium issues in discussion with @sklein94 and also minor things.

**Comment by ppxl** [comment]
Oh, one more thing. The `go.mod` file didn't work out for me straight away. I needed to call `go mod tidy && go mod vendor`

Then I observed an old version of the gRPC package for `github.com/golang/protobuf v1.5.3 // indirect` in the go.mod. IntelliJ suggests a higher version but my educated guess is that the two main dependencies 
```
	google.golang.org/grpc v1.61.1
	google.golang.org/protobuf v1.32.0
```
need an update. Especialls `google.golang.org/protobuf v1.32.0` seems to contain a high profile CVE-2024-24786.

File: grpc-protobuf/logging.proto [hunk 4]
```diff
@@ -40,4 +44,18 @@ message DoguLogMessage {
...
   // message contains the content of logged message
   string message = 2;
+}
+
+enum LogLevel {
+  DEBUG = 0;
+  INFO = 1;
+  WARN = 2;
+  ERROR = 3;
+}
```
[the remark this answers is not in the document]
Comment (nfranzeck) [reply grpc-protobuf/logging.proto]: I am with you, I thought about severity, but in the context of logging it is the wrong approach I think.

File: grpc-protobuf/logging.proto [hunk 5]
```diff
@@ -40,4 +44,18 @@ message DoguLogMessage {
...
+
+message LogLevelRequest {
```
[the remark this answers is not in the document]
Comment (nfranzeck) [reply grpc-protobuf/logging.proto:59]: I have added some docs.

File: grpc-protobuf/logging.proto (hunk 1 above)
Comment (nfranzeck) [reply grpc-protobuf/logging.proto]: Many thanks for the suggestion. I have implemented it.

File: grpc-protobuf/logging.proto (hunk 2 above)
Comment (nfranzeck) [reply grpc-protobuf/logging.proto:51]: Many thanks for the suggestion. I have implemented it.

File: grpc-protobuf/logging.proto (hunk 3 above)
Comment (nfranzeck) [reply grpc-protobuf/logging.proto:21]: I have also added a doc here.

**Comment by nfranzeck** [comment]
> Oh, one more thing. The `go.mod` file didn't work out for me straight away. I needed to call `go mod tidy && go mod vendor`
> 
> Then I observed an old version of the gRPC package for `github.com/golang/protobuf v1.5.3 // indirect` in the go.mod. IntelliJ suggests a higher version but my educated guess is that the two main dependencies
> 
> ```
> 	google.golang.org/grpc v1.61.1
> 	google.golang.org/protobuf v1.32.0
> ```
> 
> need an update. Especially `google.golang.org/protobuf v1.32.0` seems to contain a high profile [CVE-2024-24786](https://github.com/advisories/GHSA-8r3f-844c-mc37).

Maybe this is something, we can discuss? I have updated the dependencies and have added the go.sum, so for so good (Does this expression exist in English?!). Moreover, I have committed the vendor folder, so someone using this package, can do it without loading the dependencies from the proxy. We have discussed this in the last retrospective why we use `go mod vendor` without committing the pkgs to the repo. In the godex, I didn't find anything about it.

**Review by ppxl — APPROVED** [review]
LGTM

**Pull request merged**